src/test/java/com/exactpro/th2/codec/qfj/QFJCodecTest.java

```diff
@@ -422,4 +430,116 @@ public void decodeTest() {
+
+        fixMessage.getHeader().setField(new BeginString("FIXT.1.1"));
+        fixMessage.getHeader().setField(new SenderCompID("client"));
+        fixMessage.getHeader().setField(new TargetCompID("server"));
+        fixMessage.getHeader().setField(new MsgType("0"));
+        fixMessage.getHeader().setField(new TestReqID("testReqID")); //body tag in the header
+
+        fixMessage.setField(new OnBehalfOfCompID("onBehalfOfCompID")); //header tag in the body
+        String strFixMessage = fixMessage.toString();
+
+        byte[] bytes = fixMessage.toString().getBytes();
+
+        MessageGroup rawMessageGroup = MessageGroup.newBuilder()
+                .addMessages(AnyMessage.newBuilder()
+                        .setRawMessage(RawMessage.newBuilder()
+                                .setBody(ByteString.copyFrom(bytes))
+                                .setMetadata(RawMessageMetadata.newBuilder()
+                                        .setId(MessageID.newBuilder()
+                                                .setConnectionId(ConnectionID.newBuilder()
+                                                        .setSessionAlias("sessionAlias")
+                                                        .build())
+                                                .setDirection(Direction.SECOND)
+                                                .setSequence(11111111)
+                                                .build())
+                                        .setProtocol("FIX")
+                                        .setTimestamp(Timestamp.newBuilder()
+                                                .setSeconds(timestampSeconds)
+                                                .setNanos(timestampNano)
+                                                .build())
+                                        .build())
+                                .setParentEventId(EventID.newBuilder().setId("ID12345").build())
+                                .build())
+                        .build())
+                .build();
+
+
+        String bodyLength = strFixMessage.substring(strFixMessage.indexOf("\0019=") + 3, strFixMessage.indexOf("\001", strFixMessage.indexOf("\0019=") + 1));
+        String checksumValue = strFixMessage.substring(strFixMessage.lastIndexOf("\00110=") + 4, strFixMessage.lastIndexOf("\001"));
+
+        IllegalStateException thrown = assertThrows(IllegalStateException.class, () -> {
+            codec.decode(rawMessageGroup);
+        });
+
+        assertTrue(thrown
+                .getCause() //IllegalStateException: Cannot decode raw message
+                .getCause() //IllegalStateException: Cannot decode parsed message
+                instanceof FieldException); //FieldException: Tag specified out of required order, field=115
+
+        assertEquals(thrown.getCause().getCause().getMessage(), "Tag specified out of required order, field=115");
+
+        //Disabled validateFieldsOutOfOrder
```

Comment:
Extract to separate method to execute the second part when the fist part is failed